core/trino-main/src/main/java/io/trino/execution/TaskManagerConfig.java

```diff
@@ -66,7 +69,7 @@
     private Duration infoUpdateInterval = new Duration(3, TimeUnit.SECONDS);
 
     private int writerCount = 1;
-    private int taskConcurrency = 16;
+    private int taskConcurrency = nextPowerOfTwo(getAvailablePhysicalProcessorCount());
```

Comment:
is there a need for this to be a power of two?

see @tooptoop4 's https://trinodb.slack.com/archives/CP1MUNEUX/p1630586605050300

Reply:
taskConcurrency = #CPUs is optimal for single query load, right?

for high concurrency, or for more complex queries, fixed task concurrency may perform better?

Reply:
> is there a need for this to be a power of two?

We use this for number of partitions. Some local partitioning functions work only for power of 2 (we avoid modulo, which is supposedly slow)

> for high concurrency, or for more complex queries, fixed task concurrency may perform better?

I don't think this affects high concurrency much. Overall CPU time for Graviton almost didn't increase, so there isn't much overhead from higher `task_concurrency`.

I think this is also beneficial for more complex queries. Such queries consist of multiple stages. However, downstream stages feed on data produced by upstream stages. If data is not produced fast enough, then downstream stages won't have work to do too.
Of course I might be wrong, because reality is more complex

Overall observation is that other engines can better utilize CPU, so they appear good in benchmarks and tests.